### DISCUSSION TITLE
Create a commit using GitHub API v4

### DISCUSSION THREAD

**Discussion by chenleshang** [community/community#24315] [Apps, API and Webhooks]
<p>Are we able to use GitHub API to create a commit? especially v4? </p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/chenleshang">@chenleshang</a>,</p>
<p>Thank you for being here! Not at this time on APIv4, however, there is an internal issue requesting this. I’ll add this post to our Internal Feature Request list. </p>
<p>Check out our API docs for more info on <a href="https://developer.github.com/v3/git/commits/#create-a-commit" rel="nofollow noopener">how to create a commit</a> using API v3. I hope this helps!</p>
<p>Best,</p>
<p>Andrea</p>

**Comment by chenleshang** [comment]
<p>Thanks so much for reply. I think API v4 looks good and hope that we’ll get the functionality soon. </p>

**Comment by appernetic** [comment]
<p>Any progress with this?</p>

**Comment by basillatif** [comment]
<p>Hi, I am also curious about this. Is there a way to make a commit using the current API version??</p>

**Comment by rlnchow** [comment]
I answered a similar question in a different thread.
https://github.com/orgs/community/discussions/38279#discussioncomment-4143871
